qutip/solver/result.py

```diff
@@ -492,17 +492,20 @@ def _add_first_traj(self, trajectory):
             state = trajectory.final_state
             self._sum_final_states = qzero(state.dims[0])
 
-        self._sum_expect = np.array(trajectory.expect) * 0.
-        self._sum2_expect = np.array(trajectory.expect) * 0.
+        self._sum_expect = [expect * 0 for expect in trajectory.expect]
+        self._sum2_expect = [expect * 0 for expect in trajectory.expect]
```

Comment:
```suggestion
        self._sum_expect = [np.zeros_like(expect) for expect in trajectory.expect]
        self._sum2_expect = [np.zeros_like(expect) for expect in trajectory.expect]
```
Using `zeros_like` seems clearer to me.